drivers/clock_control/clock_control_rpi_pico.c

```diff
@@ -102,9 +108,16 @@
 #define AUXSRC_clk_sys   CLK_SYS
 #define AUXSRC_clk_usb   CLK_USB
 #define AUXSRC_clk_adc   CLK_ADC
-#define AUXSRC_clk_rtc   CLK_RTC
 #define AUXSRC_clk_gpin0 CLKSRC_GPIN0
 #define AUXSRC_clk_gpin1 CLKSRC_GPIN1
+#if defined(CONFIG_SOC_SERIES_RP2040)
```

Comment:
This is just a name definition based on rules, so I don't think there is any need to separate it.

Reply:
I think your comment begs the question "what's the purpose of putting this in a conditional block?"

The purpose of this is that it makes it clear that these are only used defined when working with a specific SoC series, and remove the scope for error.

After the pre-processor is run, those #defines aren't required. In contrast, if we defined them unconditionally, they'll be knocking around at the compilation stage, and someone could use them to compile something that is logically nonsense (e.g. something involving the HSTX on a RP2040).